src/wsock_message.erl

```diff
@@ -357,6 +357,8 @@ process_frame(open_close, _, Frame, Message) ->
   frame_to_complete_message(Frame, Message);
 process_frame(open_continue, _, Frame, Message) ->
   frame_for_fragmented_message(Frame, Message);
+process_frame(continue, begin_message, Frame, Message) ->
```

Comment:
> Is it possible from a wsock perspective that even the begining_message to be so fragmented that this change have a sense ?

@ctradu I don't think so.

 When you call [decode/2]() with new data this data will be tagged with the atom `begin_message`. Further in the processing chain we call [process_frames/3](https://github.com/madtrick/wsock/blob/master/src/wsock_message.erl#L321) with the frames extracted from that data. Then we analize frame by frame and we decide if we that frame belongs to a [full message](https://github.com/madtrick/wsock/blob/master/src/wsock_message.erl#L332) or a [fragmented one](https://github.com/madtrick/wsock/blob/master/src/wsock_message.erl#L330). This means that some data that was initially tagged as `begin_message` can be now tagged as `continue_message` (this is the case for fragmented messages). This means that sucessive calls to [`process_frame/2`](https://github.com/madtrick/wsock/blob/master/src/wsock_message.erl#L343) for frames belonging to fragmented messages will always have the type `continue_message`.

And when we call `process_frame/2` with the first frame of data tagged as `begin_message` the [contextualize_frame/1](https://github.com/madtrick/wsock/blob/master/src/wsock_message.erl#L389)  function can't return `continue` for frames which are the first in a message since according to the rfc the [fragmented messages must start with an opcode other than 0](https://tools.ietf.org/html/rfc6455#page-34) which means that `process_frame/4` will match for `open_continue` and not for `continue`.

I don't know if this clarifies your doubts. If not please ask again